projects/DifferentialDrive.md

```diff
@@ -0,0 +1,655 @@
+```
+
+- The PID and FF values are stored in `DriveConstants.java` and are imported. Keep in mind these are just basic values to get you started with the project.
+
+```java
+  public static final class PID {
+    public static final double P = 8.5;
+    public static final double I = 0.0;
+    public static final double D = 0.0;
+  }
+
+  public static final class FF {
+    public static final double S = 1;
+    public static final double V = 3;
+  }
+```
+
+#### Integrating PID and Feedforward
+
+Now, let’s see how these two control mechanisms work together. We’ll look at the drive method, which is responsible for controlling the motor voltages.
+
+```java
+  public void drive(double leftSpeed, double rightSpeed) {
+    final double leftFeedforward = 
+      feedforward.calculate(leftSpeed * DriveConstants.MAX_SPEED);
+    final double rightFeedforward = 
+      feedforward.calculate(rightSpeed * DriveConstants.MAX_SPEED);
+  }
+```
+
+- `MAX_SPEED` is defined in DriveConstants as such:
+
+```java
+  public static final double MAX_SPEED = 2; // Meters per second
+```
+As you know, the inputs we receive will be between -1 and 1. So by multiplying by our max speed, we can vary the range of the speed at which our robot moves. 
+
+Here, we’re calculating the feedforward voltage for each wheel based on the target speed. This helps to ensure that the motors get an initial boost that’s closer to what they need.
+
+Next, we add the PID control:
+
+```java
+    final double leftOutput = 
+      leftPIDController.calculate(leftEncoder.getVelocity(), 
+      leftSpeed * DriveConstants.MAX_SPEED);
+    final double rightOutput = 
+      rightPIDController.calculate(rightEncoder.getVelocity(), 
+      rightSpeed * DriveConstants.MAX_SPEED);
+```
+
+In this part, the PID controllers calculate how much they need to adjust the motor speed to correct any errors. They do this by comparing the actual speed (from the encoders) with the target speed.
+
+Finally, we combine the outputs and send them to the motors:
+
+```java
+      double leftVoltage = leftOutput + leftFeedforward;
+      double rightVoltage = rightOutput + rightFeedforward;
+
+      leftLeader.setVoltage(leftVoltage);
+      rightLeader.setVoltage(rightVoltage);
+```
+
+Here, the final motor voltage is calculated by adding the PID and feedforward outputs together. These voltages are then sent to the motors, controlling the movement of the robot.
+
+### Unit Testing & System Checks
+
+#### Unit Tests
+
+Think of unit tests as a way to check that each small piece of your code—like a motor or sensor—is functioning properly on its own before you start putting them all together. It's much easier to fix problems when they’re isolated to a single component than when they’re all tangled up in the full system.
+
+**Make sure to check out this [guide](https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/unit-testing.html) from WPILib on how to setup unit tests.**
```

Comment:
i hate this guide let's link the guide that I have yet to write on this (so it'll be a broken link for the moment...) instead